examples/statemachine/src/language-server/statemachine-validator.ts

```diff
@@ -27,5 +32,35 @@ export class StatemachineValidator {
             }
         }
     }
-
+    /**
+     * Checks if there are duplicate state and event names.
+     * @param statemachine the statemachine to check
+     * @param accept the acceptor to report errors
+     */
+    checkUniqueStatesAndEvents(statemachine: Statemachine, accept: ValidationAcceptor): void {
+        // check for duplicate state and event names and add them to the map
+        const names = new Map<string, State | Event | undefined>();
+        for (const state of statemachine.states) {
```

Comment:
This new approach is better, but it still suffers from a few issues:

1. The first element is littered with duplicate error messages, one for each duplicated element. However, one element should only ever raise one error.
![image](https://user-images.githubusercontent.com/4377073/224777089-020b16a5-fecc-4409-ab53-32b743afd837.png)
2. Differentiating between duplicated states and events leads to unexpected error messages, see the image above. All elements should be checked in the same loop and receive the same error message.